signer/src/storage/postgres.rs

```diff
@@ -1411,15 +1409,26 @@ impl super::DbRead for PgStore {
                 deposit_requests AS deposit_req 
                     ON deposit_req.txid = swept_deposit.deposit_request_txid
                     AND deposit_req.output_index = swept_deposit.deposit_request_output_index
-            -- TODO: The following left join is incorrect, we need to check that 
-            -- the completed deposit event is in a Stacks block that is linked 
-            -- to the canonical Bitcoin chain (#559).
             LEFT JOIN 
                 completed_deposit_events AS cde
                     ON cde.bitcoin_txid = deposit_req.txid
                     AND cde.output_index = deposit_req.output_index
-            WHERE
-                cde.bitcoin_txid IS NULL
+            LEFT JOIN
+                stacks_blocks AS cde_stacks
+                    ON cde_stacks.block_hash = cde.block_hash
+            LEFT JOIN
+                bc_blocks AS cde_bc
+                    ON cde_bc.block_hash = cde_stacks.bitcoin_anchor
+            GROUP BY
+                bc_trx.txid
+              , bc_trx.block_hash
+              , bc_blocks.block_height
+              , deposit_req.txid
+              , deposit_req.output_index
+              , deposit_req.recipient
+              , deposit_req.amount
+            HAVING
+                COUNT(cde_bc.block_hash) = 0
```

Comment:
We need the grouping to handle cases where we have two complete deposits events on two forks: just checking `IS NULL` would return the deposit because on a fork it is not confirmed (even though we have a different confirmation on the canonical chain).

Reply:
Oh I noticed this too late.

Reply:
Oh wait, we were referring to different tables maybe?

Reply:
it's the same table (the comment was for a former version of the query): since we do left joins, checking with `s_blocks.block_hash IS NULL` returns requests having a complete-deposit event in a non canonical chain, even if there's another event on the canonical one. This condition is tested in `get_swept_deposit_requests_does_not_return_deposit_requests_with_responses`: if you try removing the grouping in favor of `s_blocks.block_hash IS NULL` the test fails when asserting that we do not get requests at the end.